src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -17,12 +19,24 @@ public class QueueReceiver : BaseFunction
     private readonly CmsDbContext _db;
     private readonly ContentfulOptions _contentfulOptions;
     private readonly JsonToEntityMappers _mappers;
+    private readonly ServiceBusClient _serviceBusClient;
 
-    public QueueReceiver(ContentfulOptions contentfulOptions, ILoggerFactory loggerFactory, CmsDbContext db, JsonToEntityMappers mappers) : base(loggerFactory.CreateLogger<QueueReceiver>())
+    private const string CustomMessageProperty = "DeliveryAttempts";
+
+    private const int _defaultMaxMessageDeliveryAttempts = 4;
+    private const int _defaultMessageDeliveryDelayInSeconds = 10;
+
+    private readonly int _maxMessageDeliveryAttempts;
+    private readonly int _messageDeliveryDelayInSeconds;
+
+    public QueueReceiver(ContentfulOptions contentfulOptions, ILoggerFactory loggerFactory, CmsDbContext db, JsonToEntityMappers mappers, ServiceBusClient serviceBusClient, IConfiguration configuration) : base(loggerFactory.CreateLogger<QueueReceiver>())
     {
         _contentfulOptions = contentfulOptions;
         _db = db;
         _mappers = mappers;
+        _serviceBusClient = serviceBusClient;
+        _maxMessageDeliveryAttempts = int.TryParse(configuration["MaxMessageDeliveryAttempts"], out var configuredMaxAttempts) ? configuredMaxAttempts : _defaultMaxMessageDeliveryAttempts;
```

Comment:
Ideally these two fields (`_maxMessageDeliveryAttempts` and `_messageDeliveryDelayInSeconds`) could be their own class/struct which is injected using the [options pattern](https://learn.microsoft.com/en-us/dotnet/core/extensions/options) in the `Startup.cs`.